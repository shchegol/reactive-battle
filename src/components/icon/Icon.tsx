import React, { FC } from 'react';
import toClassNames from '@root/utils/toClassNames';
import { Props } from './types';
import './icon.scss';

/**
 * Material Icons
 * Full set on https://material.io/resources/icons
 *
 * @param {string} name - icon name
 * @param {*} rest - rest params
 * @constructor
 */

const Icon: FC<Props> = ({
  // eslint-disable-next-line react/prop-types
  name,
  ...rest
}) => (
  <i
    className={toClassNames(
      'material-icons',
      rest.className,
    )}
  >
    {name}
  </i>
);

export default Icon;
