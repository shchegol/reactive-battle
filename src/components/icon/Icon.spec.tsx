import React from 'react';
import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Icon from './index';

configure({ adapter: new Adapter() });

describe('Icon', () => {
  const snapshoot = (name: string) => {
    const wrapper = render(<Icon name={name} />);
    it('Render', () => {
      expect(wrapper).toMatchSnapshot();
    });
  };

  snapshoot('TestIcon');
});
