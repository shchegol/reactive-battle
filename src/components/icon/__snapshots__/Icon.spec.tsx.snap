// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Render 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "material-icons",
    },
    "children": Array [
      Node {
        "data": "TestIcon",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
